### PR TITLE
Bugfixes for stub.

### PR DESCRIPTION
**Pull request by mattrobertsky**
Created lax implicit enum for looser UTR validation and made some fields optional and one non-optional.

### CODE REVIEW

File: app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala [hunk 1]
```diff
@@ -120,8 +120,9 @@ case class Details(
                   contractPacker: Boolean
                   ) {
   def isValid: Boolean = {
-    producerDetails match {
-      case Some(a) => a.producerClassification.matches("^[0-1]{1}$")
+    (producer, producerDetails) match {
+      case (true,Some(a)) => a.producerClassification.matches("^[0-1]{1}$")
+      case (true, None) => false
```
Comment (hmrclt) [inline app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala]: Consider - 
```
    producerDetails match {
      case Some(a) if producer =>
        a.producerClassification.matches("^[0-1]{1}$")
      case _ => ! producer
    }
```

File: app/uk/gov/hmrc/softdrinksindustrylevystub/models/package.scala [hunk 2]
```diff
@@ -77,4 +78,11 @@ package object models {
     }
   }
 
+  private[models] implicit class ValidationOptionLitresProduced(lp: Option[LitresProduced]) {
+    def isValid: Boolean = lp match {
+      case Some(a) => a.isValid
+      case _ => true
```
Comment (hmrclt) [inline app/uk/gov/hmrc/softdrinksindustrylevystub/models/package.scala]: consider - 
```
lp.map(_.isValid).getOrElse(true)
```

File: app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala [hunk 3]
```diff
@@ -231,17 +232,18 @@ object Validation {
...
+  def isValidSites(sites: Option[List[Site]]): Boolean = {
     sites match {
       case s if s.isEmpty => true
+      case s if s.get.isEmpty => true
       case _ =>
-        sites.map(s =>
+        sites.get.map(s =>
           s.siteAddress.addressDetails.isValid &&
             s.siteAddress.contactDetails.isValid &&
             s.action.matches("^[1]{1}$") &&
```
Comment (hmrclt) [inline app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala]: consider - 
```
def isSiteValid(s: Site): Boolean = ???
sites.map(_.foldLeft(true)(_ && isSiteValid(_))).getOrElse(true)
```

File: app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala [hunk 4]
```diff
@@ -21,27 +21,42 @@ import java.time.LocalDateTime
...
+    Gen.choose(1,5).flatMap {
+      n => Gen.listOfN(n, entityActionGen).sometimes
     }                                                            // entityAction
   }.map(CreateSubscriptionRequest.apply)
 
   def genCreateSubscriptionResponse: Gen[CreateSubscriptionResponse] = {
-    Gen.const(LocalDateTime.now)                             |@| // processingDate
+    Gen.const(LocalDateTime.now).map {
+      x => x + "Z"
+    }                                                        |@| // processingDate
```
Comment (hmrclt) [inline app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala]: Is this really the best way? can we not use a formatter?

File: app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala (hunk 3 above)
Comment (mattrobertsky) [reply app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala]: Yes, but intellij suggested a forall instead... 

def isValidSites(sites: Option[List[Site]]): Boolean = {
    sites.forall(_.forall(isValidSite))
  }

  def isValidSite(site: Site): Boolean = {
    Seq(
      site.siteAddress.addressDetails.isValid,
      site.action.matches("^[1]{1}$"),
      isValidTradingName(site.tradingName),
      site.newSiteRef.matches(".{1,160}")
    ) reduce (_ && _)
  }

File: app/uk/gov/hmrc/softdrinksindustrylevystub/models/package.scala (hunk 2 above)
Comment (mattrobertsky) [reply app/uk/gov/hmrc/softdrinksindustrylevystub/models/package.scala]: Changed to..

def isValid: Boolean = lp.forall(_.isValid)

File: app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala (hunk 4 above)
Comment (mattrobertsky) [reply app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala]: There really is. I struggled to find it but it is now LocalDateTime.now.atOffset(ZoneOffset.UTC)

**Pull request merged**